src/ring_theory/chain_of_divisors.lean

```diff
@@ -248,3 +313,108 @@ begin
     ←pow_dvd_iff_le_multiplicity],
   exact pow_image_of_prime_by_factor_order_iso_dvd hn hp d (pow_multiplicity_dvd _),
 end
+
+lemma multiplicity_prime_eq_multiplicity_image_by_factor_order_iso [decidable_eq (associates M)]
+  {m p : associates M} {n : associates N} (hn : n ≠ 0)
+  (hp : p ∈ normalized_factors m) (d : {l : associates M // l ≤ m} ≃o {l : associates N // l ≤ n}):
+  multiplicity p m = multiplicity ↑(d ⟨p, dvd_of_mem_normalized_factors hp⟩) n :=
+begin
+  convert le_antisymm (multiplicity_prime_le_multiplicity_image_by_factor_order_iso hp d) _,
+  suffices : multiplicity ↑(d ⟨p, dvd_of_mem_normalized_factors hp⟩) n ≤
+    multiplicity ↑(d.symm (d ⟨p, dvd_of_mem_normalized_factors hp⟩)) m,
+  { rw [d.symm_apply_apply ⟨p, dvd_of_mem_normalized_factors hp⟩, subtype.coe_mk] at this,
+    exact this },
+  classical,
+  have := (multiplicity_prime_le_multiplicity_image_by_factor_order_iso
+    (mem_normalized_factors_factor_order_iso_of_mem_normalized_factors hn
+    (by convert hp) d) d.symm),
+  rw [subtype.coe_eta] at this,
+  convert this,
+end
```

Comment:
Using `convert` to fix issues with the wrong typeclass instance being found is not really a problem, but it's nicer to avoid these issues by choosing our `letI` carefully:
```suggestion
begin
  refine le_antisymm (multiplicity_prime_le_multiplicity_image_by_factor_order_iso hp d) _,
  suffices : multiplicity ↑(d ⟨p, dvd_of_mem_normalized_factors hp⟩) n ≤
    multiplicity ↑(d.symm (d ⟨p, dvd_of_mem_normalized_factors hp⟩)) m,
  { rw [d.symm_apply_apply ⟨p, dvd_of_mem_normalized_factors hp⟩, subtype.coe_mk] at this,
    exact this },
  letI := classical.dec_eq (associates N),
  simpa only [subtype.coe_eta] using
    (multiplicity_prime_le_multiplicity_image_by_factor_order_iso
      (mem_normalized_factors_factor_order_iso_of_mem_normalized_factors hn hp d) d.symm),
end
```